client/src/modules/entities/entities.js

```diff
@@ -0,0 +1,122 @@
+angular.module('bhima.controllers')
+  .controller('EntityController', EntityController);
+
+EntityController.$inject = [
+  'EntityService', 'ModalService', 'NotifyService', 'uiGridConstants', '$state',
+];
+
+/**
+ * Entity Controller
+ *
+ * This controller is responsible of handling entities
+ */
+function EntityController(Entity, ModalService, Notify, uiGridConstants, $state) {
+  const vm = this;
+
+  // bind methods
+  vm.deleteEntity = deleteEntity;
+  vm.editEntity = editEntity;
+  vm.createEntity = createEntity;
+  vm.toggleFilter = toggleFilter;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      {
+        field : 'display_name',
+        displayName : 'ENTITY.NAME',
+        headerCellFilter : 'translate',
+      },
+      {
+        field : 'phone',
+        displayName : 'ENTITY.PHONE',
+        headerCellFilter : 'translate',
+      },
+      {
+        field : 'email',
+        displayName : 'ENTITY.EMAIL',
+        headerCellFilter : 'translate',
+      },
+      {
+        field : 'address',
+        displayName : 'ENTITY.ADDRESS',
+        headerCellFilter : 'translate',
+      },
+      {
+        field : 'translation_key',
+        displayName : 'ENTITY.TYPE.LABEL',
+        headerCellFilter : 'translate',
```

Comment:
Can we pre-translate everything when it gets downloaded?  That might help with the grid's performance.